### PR TITLE
keep mid unchange after migration for subscribed track

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: pkg/sfu/receiver.go [hunk 1]
```diff
@@ -258,11 +258,6 @@ func (w *WebRTCReceiver) SetRTT(rtt uint32) {
 	}
 }
 
-func (w *WebRTCReceiver) SetTrackMeta(trackID livekit.TrackID, streamID string) {
```
Comment (cnderrauber) [inline pkg/sfu/receiver.go:261]: unused function

File: pkg/rtc/transport.go [hunk 2]
```diff
@@ -1164,43 +1184,83 @@ func (t *PCTransport) initPCWithPreviousAnswer(previousAnswer webrtc.SessionDesc
...
-		if err := t.initPCWithPreviousAnswer(*t.previousAnswer); err != nil {
+		if senders, err := t.initPCWithPreviousAnswer(*t.previousAnswer); err != nil {
 			t.params.Logger.Errorw("initPCWithPreviousAnswer failed", err)
 			t.lock.Unlock()
 
 			if onNegotiationFailed := t.getOnNegotiationFailed(); onNegotiationFailed != nil {
 				onNegotiationFailed()
 			}
 			return
+		} else if offer != nil {
```
Comment (cnderrauber) [inline pkg/rtc/transport.go:1225]: old clients won't send offer, keep current behavior

**Review by boks1971 — APPROVED** [review]
Don't fully understand the magic, but it does look like magic!

**Review by davidzhao — COMMENTED** [review]
nice work!

**Pull request merged**